### PR TITLE
[SIE-36] add Constrain component

### PR DESCRIPTION
**Pull request by alecherryy**

### CODE REVIEW

**Review by quanganhhoang — APPROVED** [review]
LGTM. Just a few requests for more comments.

File: src/stories/components/Constrain/Constrain.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import './styles.scss';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Component for constrain element.
+ *
```
[the remark this answers is not in the document]
Comment (alecherryy) [reply src/stories/components/Constrain/Constrain.js:8]: @quanganhhoang Explanation is on the ticket: [https://cs6510.atlassian.net/browse/SIE-36](https://cs6510.atlassian.net/browse/SIE-36)

> As a front end developer, I want to create a Constraint component to help laying out content on the page. There are three types of constraints:
> 1. Small - 600px
> 2. Default - 1200px
> 3. Wide - 1580px

> Constraints are used to enforce a max width on specific content; they should be responsive and centered.

Fundamentally, a constrain is a container `div` to help laying out pages.

File: src/stories/components/Button/Button.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const Button = ({ modifierClasses, url, text }) => {
     return (
         <a
             href={url}
-            className={['button', `${modifierClasses}`].join(' ')}
+            className={['button', `${modifierClasses}`].join(' ').trim()}
```
[the remark this answers is not in the document]
Comment (alecherryy) [reply src/stories/components/Button/Button.js:21]: Yes! A sample class output would be `button button--secondary`, where `button--secondary` is the modifier class to update and/or override styles.

File: src/stories/components/Constrain/Constrain.stories.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+const Template = (args) => {
+    return (
+        <Constrain {...args} />
+    );
+};
+
+// Default Constrain
+export const Default = Template.bind({});
```
[the remark this answers is not in the document]
Comment (alecherryy) [reply src/stories/components/Constrain/Constrain.stories.js:20]: Per Storybook's documentation:
> The template is reused across stories. Template.bind({}) makes a copy of the function which reduces code duplication. Similarly,...Primary.args makes a copy of the data, reducing data duplication.

Documentation: [https://storybook.js.org/docs/react/writing-stories/introduction](https://storybook.js.org/docs/react/writing-stories/introduction)

File: src/stories/components/Constrain/Constrain.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (quanganhhoang) [reply src/stories/components/Constrain/Constrain.js:8]: Ah right, I should have looked here. Let me find a way to link Jira tasks to PR when I have time.

**Pull request merged**